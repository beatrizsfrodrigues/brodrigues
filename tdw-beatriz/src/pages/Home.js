// src/pages/Home.js
import React from "react";

const Home = () => {
  return (
    <div>
      <h1>Welcome to the Home Page!</h1>
      <p>This is the home component of your application.</p>
      <button>Get Started</button>
    </div>
  );
};

export default Home;
