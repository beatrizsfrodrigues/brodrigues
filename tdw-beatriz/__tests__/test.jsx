import React from "react";
import { render, screen } from "@testing-library/react";
import Home from "../src/pages/Home";
import "@testing-library/jest-dom";

test("renders Home component with correct content", () => {
  render(<Home />);
  const headingElement = screen.getByRole("heading", {
    name: /welcome to the home page!/i,
  });
  expect(headingElement).toBeInTheDocument();

  const paragraphElement = screen.getByText(
    /this is the home component of your application./i,
  );
  expect(paragraphElement).toBeInTheDocument();

  const buttonElement = screen.getByRole("button", { name: /get started/i });
  expect(buttonElement).toBeInTheDocument();
});
