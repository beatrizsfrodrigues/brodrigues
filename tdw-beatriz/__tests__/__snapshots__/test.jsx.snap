// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MyComponent correctly 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Welcome to the Home Page!
    </h1>
    <p>
      This is the home component of your application.
    </p>
    <button>
      Get Started
    </button>
  </div>
</DocumentFragment>
`;
